src/Compilers/CSharp/Portable/Symbols/CollectionArgumentsSignatureOnlyMethodSymbol.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A method symbol that represents a signature for collection expression arguments.
+    /// The compiler is free to emit any equivalent method call or initialization.
+    /// </summary>
+    internal sealed class CollectionArgumentsSignatureOnlyMethodSymbol : MethodSymbol
+    {
+        internal readonly MethodSymbol WellKnownConstructor;
+
+        internal CollectionArgumentsSignatureOnlyMethodSymbol(
+            MethodSymbol wellKnownConstructor,
+            string name,
+            Symbol containingSymbol,
+            ImmutableArray<(string Name, TypeWithAnnotations Type)> parameters,
```

Comment:
Can we assert that these parameters are in sync with `wellKnownConstructor`s parameters?

Reply:
The `wellKnownConstructors` method is a definition while the types of `parameters` are potentially substituted types where the type arguments are from the call-site, so for now, I've simply checked `Length`.